Author: Alex Băluț alexandru.balut@gmail.com
Source: https://github.com/GNOME/pitivi/commit/7cc8df315d850ce4f1544b74775c79c06f0c0d30

With latest GStreamer, WaveformPreviewer.do_post_message is triggered
before the constructor finishes:

>   File "/root/pitivi/pitivi/timeline/previewers.py", line 214, in do_post_message
>     if not self.passthrough and \
> 'WaveformPreviewer' object has no attribute 'passthrough'

--- a/pitivi/timeline/previewers.py	2019-09-26 14:45:01.186046074 +0000
+++ b/pitivi/timeline/previewers.py	2018-08-29 14:33:20.000000000 +0000
@@ -224,9 +224,9 @@
 
     # pylint: disable=arguments-differ
     def do_post_message(self, message):
-        if not self.passthrough and \
-                message.type == Gst.MessageType.ELEMENT and \
-                message.src == self.level:
+        if message.type == Gst.MessageType.ELEMENT and \
+                message.src == self.level and \
+                not self.passthrough:
             struct = message.get_structure()
             peaks = None
             if struct:
